pkg/types/validation/installconfig.go

```diff
@@ -458,6 +459,99 @@ func validateClusterNetwork(n *types.Networking, cn *types.ClusterNetworkEntry,
 	return allErrs
 }
 
+func validateNetworkingClusterNetworkMTU(c *types.InstallConfig, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if c.Networking == nil {
+		return nil
+	}
+
+	if c.Networking.ClusterNetworkMTU == 0 {
+		return nil
+	}
+
+	if c.Platform.Name() != aws.Name {
+		return append(allErrs, field.Invalid(fldPath, c.Networking.ClusterNetworkMTU, "custom cluster network MTU is allowed only in AWS deployments"))
+	}
+
+	network := c.NetworkType
+	mtu := c.Networking.ClusterNetworkMTU
+
+	// higherLimitMTUVPC is the MTU limit for VPC.
+	// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/network_mtu.html#jumbo_frame_instances
+	higherLimitMTUVPC := uint32(9001)
+
+	// lowerLimitMTUVPC is the lower limit to prevent users setting too low values impacting in the
+	// cluster network performance. Tested values with 1100 decreases 70% in the network performance:
+	lowerLimitMTUVPC := uint32(1000)
+
+	// higherLimitMTUEdge defines the maximium generally supported MTU in Local and Wavelength Zones.
+	// Mostly AWS Local or Wavelength zones have limited MTU between remote
+	// zones and in the Region. It is required to warn the user when the user-defined
+	// MTU is higher than general supported.
+	// https://docs.aws.amazon.com/local-zones/latest/ug/how-local-zones-work.html#considerations
+	// https://docs.aws.amazon.com/wavelength/latest/developerguide/how-wavelengths-work.html
+	higherLimitMTUEdge := uint32(1300)
+
+	// MTU overhead for OVNKubernetes and OpenShiftSDN network plugins.
+	// https://docs.openshift.com/container-platform/4.14/networking/changing-cluster-network-mtu.html#mtu-value-selection_changing-cluster-network-mtu
+	minOverheadOVN := uint32(100)
+	minOverheadSDN := uint32(50)
```

Comment:
Are these values supposed to be constants? If so, it'd be good to declare them as such

Reply:
@r4f4 I will declare as a const but in the scope of the function, I don't want to expose those numbers to be consumed externally by other implementations as it is very specifically for this use case, mainly as those values (overhead for network plugins) are officially documented but there are bugs opened to review it.
https://issues.redhat.com/browse/OCPBUGS-14938

Reply:
perfect, thanks for the feedback. Applied!